### ISSUE TITLE
Postgres unique index is incorrect - can't enqueue the same payload in two different queues.

### ISSUE THREAD

**Issue by pconstantinou** [acaloiaro/neoq#119]
Re:
https://github.com/acaloiaro/neoq/blob/main/backends/postgres/migrations/20230902105301_make_fingerprint_index_unique.up.sql#L2

The index would prevent two separate queues from enqueuing the same payload.

It should be:
```
CREATE UNIQUE INDEX IF NOT EXISTS neoq_jobs_fingerprint_unique_idx ON neoq_jobs (queue, fingerprint, status) WHERE NOT (status = 'processed');
```

**Comment by acaloiaro** [comment]
Right you are! This looks pretty dumb staring me in the face :)

**Comment by acaloiaro** [comment]
Resolved by #122
